src/Microsoft.Sbom.Parsers.Spdx22SbomParser/Parser/ParserUtils.cs

```diff
@@ -228,6 +228,12 @@ public static void GetMoreBytesFromStream(Stream stream, ref byte[] buffer, ref
 
             leftover.CopyTo(buffer);
             bytesRead = stream.Read(buffer.AsSpan(leftover.Length));
+
+            // stream.Read doesn't always return the whole buffer length, so we need to fill the rest
```

Comment:
If we expect to fill the entire buffer, would it be preferable to use [ReadExactly](https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.readexactly?view=net-7.0#system-io-stream-readexactly(system-span((system-byte)))) instead, which according to doc advances the position until stream is filled?

Reply:
As discussed offline, I'd be super here for that except that `ReadExactly` was introduced in net7 and doesn't return a "bytesRead" count, so we'd have to upgrade this project to net6 and also restructure a bit.